File: Curiosity/speed_benchmarking/read_file.java
import java.io.*;
import java.util.*;

class ReadFile{

    public static void main(String [] args){
        if (args.length < 1){
            System.out.println("Usage: java ReadFile <filename>");
            System.exit(0);
        }

        String fileName = args[0];
        System.out.format("Reading : %s%n", fileName);

        long start = System.currentTimeMillis();

        Vector <String> content = new Vector<>();
        BufferedReader br = null;
        String data = "";
        try{
            br = new BufferedReader(new FileReader(fileName));
            String line = "";
            while( (line = br.readLine()) != null){
                content.add(line);
            }
        } catch(FileNotFoundException e){System.out.println("File Not Found!");}
          catch(IOException e){e.printStackTrace();}

        long end = System.currentTimeMillis();
        System.out.format("%d Lines Read.    %d ms.%n", content.size(), (end-start) );
    }

}

File: Curiosity/speed_benchmarking/read_file.py
import time
import sys


if len(sys.argv) < 2:
    print 'Usage: python read_file.py <file_name>'
    exit(0)
print 'Reading: %s' % sys.argv[1]
t0 = time.time()
data = list()
[data.append(line.replace('\n', '')) for line in open(sys.argv[1], 'r').readlines()]

print '%d Lines Read\t %fs Elapsed' % (len(data), time.time() - t0)
